src/libraries/System.Private.CoreLib/src/System/String.cs

```diff
@@ -447,6 +447,40 @@ public unsafe void CopyTo(int sourceIndex, char[] destination, int destinationIn
                 elementCount: (uint)count);
         }
 
+        // TODO: https://github.com/dotnet/runtime/issues/51061
+        // Make these {Try}CopyTo methods public and use throughout dotnet/runtime.
+
+        /// <summary>Copies the contents of this string into the destination span.</summary>
+        /// <param name="destination">The span into which to copy this string's contents.</param>
+        /// <exception cref="System.ArgumentException">The destination span is shorter than the source string.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal void CopyTo(Span<char> destination)
+        {
+            if ((uint)Length <= (uint)destination.Length)
+            {
+                Buffer.Memmove(ref destination._pointer.Value, ref _firstChar, (uint)Length);
```

Comment:
Shouldn't the length passed to memmove be in bytes, not chars?

Reply:
Not needed. Memmove is a generic method and automatically multiplies by sizeof(T).